### ISSUE TITLE
Before marking an ingress as ready perform a DNS check

### ISSUE THREAD

**Issue by maleck13** [Kuadrant/kcp-glbc#354]
## Issue Description
Once the DNSRecord resource has synced to AWS, after the TTL we should perform a DNS lookup to ensure the DNS is visible before marking the status of the ingress as ready. 

- [ ] Add a ingress ready status if not present
- [ ] Add a delayed DNSLook up on the provided ingress host to ensure we can see the DNS before marking the ingress ready

**Comment by maleck13** [comment]
@philbrookes curious on your thoughts here? I think checking DNS is available via the authoritive ns before marking an ingress as ready is a good step WDYT?

**Comment by philbrookes** [comment]
this seems like a good idea to me. Can we edit the ingress status though? Wouldn't KCP syncer immediately overwrite it with the status from the synctarget?

**Comment by maleck13** [comment]
I think it would if advnaced scheduling was not enabled currently. But once that stuff is not hidden behind that gate, status would always be in the annotation

**Comment by philbrookes** [comment]
right, but whether it's in the annotation or the status object, if changed the status from Ready (as the sync target sets it) to not ready, wouldn't it be set back to Ready by the syncer when it next syncs that object?

**Comment by maleck13** [comment]
What I mean is if the status was always in the annotation, then we can set the actual status (which wont be touched by KCP in that situation) and set the lb host to be the managed host we provide

**Comment by philbrookes** [comment]
ah yes, ok, that's a good idea yeah!
